### PR TITLE
Adds --majority option to classify using majority vote algorithm

### LINKED ISSUE #984: Add option to use majority vote instead of LCA for taxonomic classification
Hey all!

I've been playing with contig-level taxonomic classification for a while and I noticed that most of the times taking the LCA is too "pessimistic" and often results in a high-rank classification (domain, phylum, class). After some testing I've noticed that taking the majority vote usually gave me more specific taxa without over-classification.

It is important to note that I'm using the majority vote approach to go from gene-level classification to contig-level classification. As k-mer data is probably noisier, it may not work as well as it did for me.

As at the time I coulnd't find a Python implementation of the majority vote algorithm, I ended up creating a [package just for that](https://github.com/apcamargo/taxopy/blob/master/taxopy/utilities.py#L61) (@luizirber I promisse I'll try to implement it in [taxonomy](https://github.com/onecodex/taxonomy) as soon as I got the time!).

### PR DESCRIPTION
**Pull request by erikyoung85**
This PR adds command line option of `--majority` to `lca classify` which will use the **majority vote** algorithm instead of **lca**.
Fixes #984
- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by erikyoung85** [comment]
when running `c.run_sourmash('lca', 'classify', '--db', lca_db, '--query', sig1and2)`, it returns the following match:
```
ID,status,superkingdom,phylum,class,order,family,genus,species,strain
e9b8c46c,found,Bacteria,Proteobacteria,unassigned,unassigned,Alteromonadaceae,,,


loaded 1 LCA databases. ksize=31, scaled=10000 moltype=DNA
finding query signatures...
outputting classifications to -
classified 1 signatures total
```

File: sourmash/lca/command_classify.py [hunk 1]
```diff
@@ -48,11 +48,15 @@ def classify_signature(query_sig, dblist, threshold):
...
     tree = {}
-
-    for lca, count in counts.most_common():
-        if count < threshold:
-            break
-
+    lca = []
+
+    if majority_vote:
+        majority = counts.most_common()[0][0]
```
Comment (ctb) [inline sourmash/lca/command_classify.py:54]: error condition to check: what if there are no counts?

Also note, you can pull the first result off of `most_common` with `next(iter(most_common()))`.

File: tests/test_lca.py [hunk 2]
```diff
@@ -1153,6 +1153,56 @@ def test_index_and_classify_internal_unassigned_multi():
         assert 'loaded 1 LCA databases' in err
 
 
+@utils.in_tempdir
+def test_index_and_classify_internal_unassigned_multi_2(c):
```
Comment (ctb) [inline tests/test_lca.py:1157]: pls put `majority` in the test name, and add a comment as the first line of the function that says something like `test --majority-vote option`

File: tests/test_lca.py [hunk 3]
```diff
@@ -1153,6 +1153,56 @@ def test_index_and_classify_internal_unassigned_multi():
...
+    print("\nc.last_command =", c.last_command)
+    print("\nc.last_result.out =", c.last_result.out)
+    print("\nc.last_result.err =", c.last_result.err)
+
+    assert 'ID,status,superkingdom,phylum,class,order,family,genus,species' in c.last_result.out
+    assert 'nomatch' in c.last_result.out
+    assert 'classified 1 signatures total' in c.last_result.err
+    assert 'loaded 1 LCA databases' in c.last_result.err
+
+    # majority vote classify should yeild results
```
Comment (ctb) [inline tests/test_lca.py:1192]: yield instead of yeild :)

File: tests/test_lca.py [hunk 4]
```diff
@@ -1153,6 +1153,56 @@ def test_index_and_classify_internal_unassigned_multi():
...
+    c.run_sourmash('lca', 'classify', '--db', lca_db, '--query', sig1and2, '--majority_vote')
```
Comment (ctb) [inline tests/test_lca.py:1193]: can we split this test into two functions, one that tests the result _without_ `--majority` and one that tests the result with?

File: tests/test_lca.py [hunk 5]
```diff
@@ -1153,6 +1153,56 @@ def test_index_and_classify_internal_unassigned_multi():
...
+    # lca classify should yield no results
+    c.run_sourmash('lca', 'classify', '--db', lca_db, '--query', sig1and2)
+
+    print("\n\nLCA CLASSIFY:")
+    print("\nc.last_command =", c.last_command)
+    print("\nc.last_result.out =", c.last_result.out)
+    print("\nc.last_result.err =", c.last_result.err)
+
+    assert 'ID,status,superkingdom,phylum,class,order,family,genus,species' in c.last_result.out
+    assert 'nomatch' in c.last_result.out
```
Comment (ctb) [inline tests/test_lca.py:1188]: I get
```
e9b8c46c,disagree,,,,,,,,
``` on master branch for this command, but 

```
e9b8c46c,found,Bacteria,Proteobacteria,unassigned,unassigned,Alteromonadaceae,,,
```
on this branch. So, this branch is changing behavior of `lca classify` even without `--majority-vote` being specified.

**Comment by ctb** [comment]
thank you!

**Pull request merged**